File: src/main/java/com/staymate/service/BedService.java
package com.staymate.service;

import java.util.List;

import com.staymate.dto.BedDTO;
import com.staymate.entity.Bed;

public interface BedService {

    Bed addBed(BedDTO dto);

    List<Bed> getAllBeds();

    Bed getBedById(Long id);

    Bed updateBed(Long id, BedDTO dto);

    void deleteBed(Long id);

    Bed allocateBed(Long bedId, Long residentId);

    Bed vacateBed(Long bedId);

}

File: frontend/src/services/bedService.js
import api from "./api";

const API = "/beds";

const getAllBeds = () => api.get(API);

const getBed = (id) => api.get(`${API}/${id}`);

const addBed = (bed) => api.post(API, bed);

const updateBed = (id, bed) => api.put(`${API}/${id}`, bed);

const deleteBed = (id) => api.delete(`${API}/${id}`);

const allocateBed = (bedId, residentId) =>
    api.put(`${API}/allocate/${bedId}/${residentId}`);

const vacateBed = (bedId) =>
    api.put(`${API}/vacate/${bedId}`);

export default {
    getAllBeds,
    getBed,
    addBed,
    updateBed,
    deleteBed,
    allocateBed,
    vacateBed
};